src/screen/ProductPage.tsx

```diff
@@ -1,61 +1,106 @@
 import { NavBar } from "../component/NavBar";
 import { useState } from "react";
-import { useFetchCategories } from "../domain/categories";
+import { Category, useFetchCategories } from "../domain/categories";
 import {
   Product,
   useCreateProduct,
   useFetchProduct,
   useDeleteProduct,
   useEditProduct,
+  ProductRequest,
 } from "../domain/product";
+import { FetchCategories } from "./CategoryPage";
 
 type ProductPageProps = {
   HomeClick: () => void;
   CategoryClick: () => void;
   ProductClick: () => void;
 };
 
+type ProductForm = Omit<Product, "id">;
```

Comment:
type ProductForm ini bisa diimprove jadi kayak gini, biar id itu selalu ada isinya ketika mode edit : 

```suggestion
type ProductForm = 
 | { type: "create", selectedId: null, values: Omit<Product, "id"> }
 | { type: "edit", selectedId: number, values: Omit<Product, "id"> }
 
 const [inputProduct, setInputProduct] = useState<ProductForm>({
    type: "create",
    selectedId: null,
    values: {
       title: "",
       categoryId: 1,
       price: 0,
    },
  });
  
  
  if (inputProduct.type === "create") {
      inputProduct.selectedId // null
  } else {
    inputProduct.selectedId // number
  }
```

Reply:
updated mas